velox/vector/BaseVector.h

```diff
@@ -863,6 +863,29 @@ class BaseVector {
     storageByteCount_ = std::nullopt;
   }
 
+  /// Slice a buffer with specific type.
+  /// For boolean type and if the 'offset' is not multiple of 8, return a
+  /// shifted copy, new buffer is allocated from 'pool'.
+  /// Otherwise return a BufferView into the original buffer (with shared
+  /// ownership of original buffer).
+  ///
+  /// @param type The data type of the elements in new buffer. Must be primitive
+  /// type or OPAQUE type.
+  /// @param buf A pointer to the buffer to be sliced. If this is null, the
+  /// function returns a null pointer.
+  /// @param offset The starting point in the buffer from which the slice
+  /// begins. Must be non-negative.
+  /// @param length The number of elements to include in the slice. Must be
```

Comment:
It was strange for me to see an API that operates on a Buffer and takes Type. I just realized that this is inconsistent with other Buffer APIs, i.e. `AlignedBuffer::allocate<T>` and `AlignedBuffer::reallocate<T>`.

I feel it is better to move this API next to other Buffer APIs and change it from taking Type parameter to take T template parameter.

`Buffer::slice<T>(offset,...)`

Then, offset and length will be in bytes, not elements.

Reply:
> I feel it is better to move this API next to other Buffer APIs and change it from taking Type parameter to take T template parameter.

Sure.

> Then, offset and length will be in bytes, not elements.

In both `AlignedBuffer::allocate<T>` and `AlignedBuffer::reallocate<T>`, we pass number of elements not bytes. Should we align this API with them?
Thanks.

```
  static BufferPtr allocate(
      size_t numElements,
      velox::memory::MemoryPool* pool,
      const std::optional<T>& initValue = std::nullopt) 
```

Reply:
> In both AlignedBuffer::allocate<T> and AlignedBuffer::reallocate<T>, we pass number of elements not bytes. Should we align this API with them?

@zhli1142015 You are right. I forgot. Yes, let's make this new API consistent with the existing ones. Thanks.

Reply:
Updated, please help to check, thanks.